lib/get-age.js

```diff
@@ -0,0 +1,154 @@
+  });
+  return Promise.resolve(issues);
+};
+
+/**
+ * Calculate the age of the issue.
+ * If the issue is closed,
+ * the age is the duration between closedAt date and the createdAt date.
+ * If the issue is open,
+ * the age is the duration between current date and the createdAt date
+ * @param {Object[]} issues
+ */
+function calculateAge(issues) {
+  issues.forEach(function(issue){
+    var lapse;
+    if (issue.closedAt) {
+      lapse = new Date(issue.closedAt - issue.createdAt);
+    } else {
+      lapse = new Date() - new Date(issue.createdAt);
+    }
+    var one_day = 1000 * 60 * 60 * 24;
+    issue.age = Math.round(lapse / one_day);
+  });
+};
+
+/**
+ * Build the output JSON with the desired attributes
+ * @param {Object[]} issues
+ */
+function filterIssueAttr(issues) {
+  var filteredIssues = [];
+  issues.forEach(function(issue){
+    var newIssue = {};
+    newIssue.number = issue.number;
+
+    // make sure there is no comma to affect the csv conversion result
+    newIssue.title = issue.title.replace(',', '_');
+
+    newIssue.htmlUrl = issue.htmlUrl;
+    newIssue.age = issue.age;
+    newIssue.state = issue.state;
+    newIssue.createdAt = issue.createdAt;
+    newIssue.closedAt = issue.closedAt;
+    filteredIssues.push(newIssue);
+  });
+  return Promise.resolve(filteredIssues);
+}
+
+exports.handler = function(argv, options) {
+  var opts = options || {};
+  var _console = opts.console || console;
+  // Shim the console log call with our own.
+  _console.lg = _console.log;
+  _console.log = function() {
+    return argv.silent ? null : _console.lg.apply(this, arguments);
+  };
+
+  // set verbose to true all the time in order to get the right result
+  // argv.verbose = true;
+  var octo = util.octokatBuilder(argv, opts);
+  debug('Finding issues on %s...', argv.repo);
+  var target = util.parseGitRepo(argv.repo);
+  debug('Owner: %s Repository: %s', target.owner, target.repo);
+  _console.log('Searching for issues...');
+  var issueFinder = new IssueFinder({
+    octo: octo,
+    owner: target.owner,
+    repo: target.repo,
+  });
+
+  var findOptions = {};
+  if (argv.state) {
+    findOptions.state = argv.state;
+  }
+  return issueFinder.find(findOptions).then(function(issues) {
+  // return issueFinder.find({ state: argv.status }).then(function(issues) {
+    debug('Matching issues returned: %s', issues.length);
+
+    getIssuesOnly(issues).then(function(issues){
+      calculateAge(issues);
+      filterIssueAttr(issues).then(function(issues){
+        if (argv.count) {
+          _console.lg('Found %s matching issues. ', issues.length);
+        } else {
+          _console.lg(JSON.stringify(issues));
+        }
+
+        if (argv.outfile) {
+          // convert the issues into csv
+          var opts = {
+            data: issues,
+            quotes: '',
+            excelString: true,
+          };
+          var csv = json2csv(opts);
+          fs.writeFile(argv.outfile, csv, function(err){
```

Comment:
You can use a Promisified version of this with Bluebird.
```
// In the imports declarations...
var fs = require('fs');
var writeFile = Promise.promisify(fs.writeFile);

// Down below in the code...
return writeFile(argv.outFile, csv);
```

This will also have the added benefit of being handled by the `.catch` statement below and logged accordingly without duplicate calls to `_console.lg`.